src/Transportation.cpp

```diff
@@ -0,0 +1,125 @@
+#include "../include/Transportation.h"
+
+// Destructor
+
+Transportation::~Transportation()
+{
+    cout << "Transportation" << type << " deleted" << endl;
+}
+
+// Constructors
+
+Transportation::Transportation()
+{
+    type = "";
+    origin = Location();
+    destination = Location();
+    departureTime = Time();
+    arrivalTime = Time();
+    price = 0;
+    passengers = 0;
+}
+
+Transportation::Transportation(string type, Location origin, Location destination, Time departureTime, Time arrivalTime, double price, int passengers)
+{
+    this->type = type;
+    this->origin = origin;
+    this->destination = destination;
+    this->departureTime = departureTime;
+    this->arrivalTime = arrivalTime;
+    this->price = price;
+    this->passengers = passengers;
+}
+
+// Getters
+
+string Transportation::getType() const { return type; }
+
+Location Transportation::getOrigin() const { return origin; }
+
+Location Transportation::getDestination() const { return destination; }
+
+Time Transportation::getDepartureTime() const { return departureTime; }
+
+Time Transportation::getArrivalTime() const { return arrivalTime; }
+
+double Transportation::getPrice() const { return price; }
+
+int Transportation::getPassengers() const { return passengers; }
+
+// Setters
+
+bool Transportation::setType(string t)
+{
+    if (t.empty())
+    {
+        return false;
+    }
+    type = t;
+    return true;
+}
+
+bool Transportation::setOrigin(Location o)
+{
+    if (o.getLongitude() == destination.getLongitude() && o.getLatitude() == destination.getLatitude())
+    {
+        return false;
+    }
+    origin = o;
+    return true;
+}
+
+bool Transportation::setDestination(Location d)
+{
+    if (d.getLongitude() == origin.getLongitude() && d.getLatitude() == origin.getLatitude())
```

Comment:
you can use distanceTo method from Location class. if distance == 0, return false ((Optional))